wire-compiler/src/main/java/com/squareup/wire/java/TypeWriter.java

```diff
@@ -661,9 +665,21 @@ private MethodSpec builderBuild(WireMessage message, ClassName javaType) {
         .addAnnotation(Override.class)
         .addModifiers(PUBLIC)
         .returns(javaType);
-    if (message.hasRequiredFields()) {
-      result.addStatement("checkRequiredFields()");
+
+    List<WireField> requiredFields = message.getRequiredFields();
+    if (!requiredFields.isEmpty()) {
+      List<String> conditionals = new ArrayList<String>();
+      List<String> missingArgs = new ArrayList<String>();
+      for (WireField requiredField : requiredFields) {
+        conditionals.add(requiredField.name() + " == null");
+        missingArgs.add(requiredField.name() + ", \"" + requiredField.name() + '"');
```

Comment:
Then you could use `$S`, which will Do The Right Thing if requiredField.name() needs any escaping.